### DISCUSSION TITLE
How to get the "channel_input" attribute

### DISCUSSION THREAD

**Discussion by chatasos** [scrapli/nornir_scrapli#145] [Q&A] [answered]
Hi,

In scrapli response i have also **channel_input** (plus **elapsed_time** and some other attributes) returned.

i.e.
```
with IOSXEDriver(**device) as conn:
	response = conn.send_command("show ip int brief")
	print(response.channel_input)
	print(response.result)
```

I'm newbie here and trying to find a way to get it included in the nornir plugin response too.
Is this somehow accessible or do i need to write my own code?

**Comment by carlmontanari** [comment, marked as answer]
yep, can get the "normal" scrapli response like [this](https://github.com/scrapli/nornir_scrapli/blob/main/examples/structured_data/demo.py#L30-L49)

**Comment by chatasos** [comment]
Althought i had already seen that example, initially i didn't manage to make it work.
After a little more research i found the solution:

```

  command_results = nr.run(task=send_commands_from_file, file="commands_file.txt", strip_prompt=False)

  for device in command_results:
    for original_response in command_results[device][0].scrapli_response:
      print(device, ":", original_response.channel_input)

```

```
router1 : show version
router1 : show snmp
router2 : show version
router2 : show snmp
router3 : show version
router3 : show snmp
```

Thank you very much for your quick answer!

Btw, you may want to add an option/parameter to skip processing null commands (i.e. empty lines, when provided through a file), so you don't get null entries in the above structure.
